flytekit/tools/fast_registration.py

```diff
@@ -3,27 +3,42 @@
 import gzip
 import hashlib
 import os
+import pathlib
 import posixpath
 import subprocess
+import sys
 import tarfile
 import tempfile
 import typing
 from dataclasses import dataclass
+from enum import Enum
 from typing import Optional
 
 import click
+from rich import print as rich_print
+from rich.tree import Tree
 
 from flytekit.core.context_manager import FlyteContextManager
 from flytekit.core.utils import timeit
 from flytekit.exceptions.user import FlyteDataNotFoundException
 from flytekit.loggers import logger
 from flytekit.tools.ignore import DockerIgnore, FlyteIgnore, GitIgnore, Ignore, IgnoreGroup, StandardIgnore
-from flytekit.tools.script_mode import tar_strip_file_attributes
+from flytekit.tools.script_mode import _filehash_update, _pathhash_update, ls_files, tar_strip_file_attributes
 
 FAST_PREFIX = "fast"
 FAST_FILEENDING = ".tar.gz"
 
 
+class CopyFileDetection(Enum):
+    LOADED_MODULES = 1
+    ALL = 2
+    # This is a temporary option to be removed in the future. In the future this value of the enum should simply
+    # be Python None. Here now to distinguish between users explicitly setting --copy none and not setting the flag.
+    # This is only used for register, not for package or run because run doesn't have a no-fast-register option and
+    # package is by default non-fast.
+    TEMP_NO_COPY = 3
```

Comment:
I think it's better to keep this enum and not use Python `None`. This way `CopyFileDetection` fully describes all the different copy modes.

Reply:
sounds good.  I will update the comment and rename the value.